Source/pfile.cpp

```diff
@@ -225,6 +226,8 @@ bool pfile_ui_set_hero_infos(bool (*ui_add_hero_info)(_uiheroinfo *))
 
 				UnPackPlayer(&pkplr, 0, false);
 
+				pfile_SFileCloseArchive(archive);

```

Comment:
Let's change the argument to `pfile_SFileCloseArchive` to `HANDLE *` and make `pfile_SFileCloseArchive` set it to NULL instead